content/news/2023-06-from-single-files-to-collections/index.md

```diff
@@ -0,0 +1,180 @@
+
+If an element of a collection is failed or empty the entire downstream processing is stopped, which can be rather annoying if one want to process a large amount of data and got stuck due to a few elements. Two solutions are proposed to handle such cases.
+
+## Intermediate workflow specific solution
+
+### Filter the collection
+
+Collections can be filtered for failed or empty datasets using collection tools such as [filter empty datasets](https://usegalaxy.eu/?tool_id=__FILTER_EMPTY_DATASETS__&version=1.0.0) and [filter failed datasets](https://usegalaxy.eu/?tool_id=__FILTER_FAILED_DATASETS__&version=1.0.0). 
+
+<div class="center">
+<div class="img-sizer" style="width: 100%">
+
+![Empty collection](./figs/empty_collection.png)
+
+</div>  
+<figcaption>
+  Filter failed collection elements
+</figcaption>
+</div>
+
+<div class="center">
+<div class="img-sizer" style="width: 100%">
+
+![Empty collection](./figs/failed_collection.png)
+
+</div>  
+<figcaption>
+  Filter empty collection elements
+</figcaption>
+</div>
+
+
+Although this can solve the issue in some cases immediately, further considerations need to be made. 
+First, often one cannot really know at which step the collection will be affected.
+To cover all cases one would need to add the filter steps for every produced collection, which will increase the workflow steps unreasonably. 
+Secondly, the filtering will change the size of the collection. If downstream tools depend on a specific collection size, which is always the case if a tool
+takes two or more collections as input, the tool will also fail. That's basically a follow up problem of the first problem.
+This case can still be solved by a intermediate step where the second collection is basically reindexed by the same element identifiers then the collection with
+missing elements. 
+
+<div class="center">
+<div class="img-sizer" style="width: 100%">
+
+![Empty collection](./figs/reindex_collection.png)
+
+</div>  
+<figcaption>
+  Proposed solution to reindexed second collection using elements of the first collection.
+</figcaption>
+</div>  
+
+However, if this logic needs to be applied to multiple collections or again for multiple steps, the workflow becomes even more unreasonably large and complex. 
+
+Furthermore it needs to be considered, that filtering empty or failed elements from a collection
+can hide the root cause of the problem. The question why some tool runs fail or produce empty output should always be further investigated.
+
+### Rerun the workflow with rerun activated 
+
+In some cases a rerun of the workflow can also solve the issue, for example in our use case where the failure was rather randomly.
+For this strategy it is beneficial to activate the rerun option of galaxy (User/Preferences/Manage Information) on instance with version > 23.0. 
```

Comment:
Is this meant to refer to the job cache ? re-running single elements has been possible for many years.

Reply:
Is this not the cache option: "Do you want to be able to re-use equivalent jobs ?" Thats what I found on .org ... or did you mean another option ?

Reply:
`Do you want to be able to re-use equivalent jobs` -> this is the job cache.
I don't see how that is related to `Rerun the workflow with rerun activated` ?